djitellopy/tello.py

```diff
@@ -1057,12 +1060,39 @@ def update_frame(self):
         """
         try:
             for frame in self.container.decode(video=0):
-                self.frame = np.array(frame.to_image())
+                with self.lock:
+                    self.frames.put(np.array(frame.to_image()))
```

Comment:
The `Queue` implementation should be thread safe by itself and should not require an outer mutex for getting/putting